src/compiler/scala/tools/nsc/Reporting.scala

```diff
@@ -114,11 +114,12 @@ trait Reporting extends scala.reflect.internal.Reporting { self: ast.Positions w
 
       val msg = s"$featureDesc $req be enabled\nby making the implicit value $fqname visible.$explain" replace ("#", construct)
       // don't error on postfix in pre-0.13.18 xsbt/Compat.scala
-      def isSbtCompat =
-        (featureName == "postfixOps" && pos.source.path.endsWith("/xsbt/Compat.scala") && Thread.currentThread.getStackTrace.exists(_.getClassName.startsWith("sbt.")))
-      if (required && !isSbtCompat) {
-        reporter.error(pos, msg)
-      } else featureWarning(pos, msg)
+      def isSbtCompat = (featureName == "postfixOps"
+        && pos.source.file.name == "Compat.scala"
+        && pos.source.file.file.getParentFile.getName == "xsbt"
```

Comment:
I like this implementation better, but `pos.source.file.file` and `getParentFile` could return null here.

Reply:
yeah but the API doesn't tell me that. At least it would have to pass the Compat.scala test first.